### ISSUE TITLE
0.10.0-rc1 query fails with: This method should not be invoked for this granularity type

### ISSUE THREAD

**Issue by erikdubbelboer** [druid-io/druid#4097]
A query (see below) that worked with previous versions now fails on 0.10.0-rc1 with:
```
2017-03-23T05:06:59,423 ERROR [processing-5] io.druid.query.groupby.epinephelinae.GroupByMergingQueryRunnerV2 - Exception with one of the sequences!
java.lang.UnsupportedOperationException: This method should not be invoked for this granularity type
        at io.druid.java.util.common.granularity.DurationGranularity.toDateTime(DurationGranularity.java:117) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.segment.QueryableIndexStorageAdapter$CursorSequenceBuilder$1.apply(QueryableIndexStorageAdapter.java:419) ~[druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.segment.QueryableIndexStorageAdapter$CursorSequenceBuilder$1.apply(QueryableIndexStorageAdapter.java:382) ~[druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.MappingAccumulator.accumulate(MappingAccumulator.java:42) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.BaseSequence.accumulate(BaseSequence.java:46) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.MappedSequence.accumulate(MappedSequence.java:43) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence$1.get(WrappingSequence.java:50) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.SequenceWrapper.wrap(SequenceWrapper.java:55) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.FilteredSequence.accumulate(FilteredSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.MappedSequence.accumulate(MappedSequence.java:43) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence$1.get(WrappingSequence.java:50) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.SequenceWrapper.wrap(SequenceWrapper.java:55) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.ConcatSequence.accumulate(ConcatSequence.java:43) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence$1.get(WrappingSequence.java:50) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.SequenceWrapper.wrap(SequenceWrapper.java:55) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.LazySequence.accumulate(LazySequence.java:40) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence$1.get(WrappingSequence.java:50) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.SequenceWrapper.wrap(SequenceWrapper.java:55) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.LazySequence.accumulate(LazySequence.java:40) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence$1.get(WrappingSequence.java:50) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.SequenceWrapper.wrap(SequenceWrapper.java:55) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.query.spec.SpecificSegmentQueryRunner$2.accumulate(SpecificSegmentQueryRunner.java:83) ~[druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence$1.get(WrappingSequence.java:50) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.query.spec.SpecificSegmentQueryRunner.doNamed(SpecificSegmentQueryRunner.java:169) ~[druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.query.spec.SpecificSegmentQueryRunner.access$200(SpecificSegmentQueryRunner.java:43) ~[druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.query.spec.SpecificSegmentQueryRunner$3.wrap(SpecificSegmentQueryRunner.java:149) ~[druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence$1.get(WrappingSequence.java:50) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.query.CPUTimeMetricQueryRunner$1.wrap(CPUTimeMetricQueryRunner.java:78) ~[druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:45) ~[java-util-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.query.groupby.epinephelinae.GroupByMergingQueryRunnerV2$1$1$1.call(GroupByMergingQueryRunnerV2.java:225) [druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at io.druid.query.groupby.epinephelinae.GroupByMergingQueryRunnerV2$1$1$1.call(GroupByMergingQueryRunnerV2.java:216) [druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
        at io.druid.query.PrioritizedListenableFutureTask.run(PrioritizedExecutorService.java:271) [druid-processing-0.10.0-rc1.jar:0.10.0-rc1]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
```

The query:

```json
{
  "aggregations": [
    {
      "aggregator": {
        "fieldName": "requests",
        "name": "impressions",
        "type": "longSum"
      },
      "filter": {
        "fields": [
          {
            "dimension": "request_type_id",
            "type": "selector",
            "value": 1
          },
          {
            "dimension": "result_type_id",
            "type": "selector",
            "value": 0
          }
        ],
        "type": "and"
      },
      "type": "filtered"
    },
    {
      "fieldName": "fooxxx",
      "name": "fooyyy",
      "type": "doubleSum"
    },
    {
      "fieldName": "barxxx",
      "name": "baryyy",
      "type": "doubleSum"
    }
  ],
  "dataSource": "requests",
  "dimensions": [
    "dimension_x",
    "dimension_y"
  ],
  "granularity": "DAY",
  "intervals": [
    "2017-03-01T00:00:00+00:00/2017-03-23T00:00:00+00:00"
  ],
  "queryType": "groupBy"
}
```

**Comment by erikdubbelboer** [comment]
Our data has a granularity of `HOUR`.

**Comment by gianm** [comment]
This looks like #4001, a bug that happens with some of the predefined granularities like "day" when your brokers are on 0.9.x and other nodes are on 0.10.0-rc1. It'll be fixed in 0.10.0-rc2.

I think the query you pasted should also work if your brokers are all running 0.10.0-rc1 too.

**Comment by erikdubbelboer** [comment]
I only switched one historical node so I don't know if all nodes running 0.10.0-rc1 would work. I'll try this next week.

**Comment by erikdubbelboer** [comment]
This is fixed in 0.10.0-rc2.

**Comment by gianm** [comment]
Thanks for confirming!
